src/openrct2/peep/Peep.cpp

```diff
@@ -2880,6 +2890,13 @@ static bool peep_interact_with_shop(Peep* peep, const CoordsXYE& coords)
         return true;
     }
 
+    // If we are queuing ignore the 'shop'
+    // This can happen when paths clip through track
+    if (peep->State == PeepState::Queuing)
```

Comment:
Shouldn't this be
```c++
 if (peep->State == PeepState::Queuing && peep->CurrentRide != rideIndex)
 ```

Like the other one above?

Reply:
So I thought about that for a while. I don't think so. It can only get to here if a track is intersecting a path. So in theory we want it to skip it even if the track is from the ride we are queueing for as we are going to the entrance only. Counter argument to this is that the ride type has already been checked to see if it's a shop. And shops don't have track or queues so it wouldn't matter having the extra check as it could never get to this point.

Reply:
Ok, sold!